pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -381,9 +384,9 @@ func (jm *ControllerV2) updateCronJob(old interface{}, curr interface{}) {
 	// if the change in schedule results in next requeue having to be sooner than it already was,
 	// it will be handled here by the queue. If the next requeue is further than previous schedule,
 	// the sync loop will essentially be a no-op for the already queued key with old schedule.
-	if oldCJ.Spec.Schedule != newCJ.Spec.Schedule {
+	if oldCJ.Spec.Schedule != newCJ.Spec.Schedule || (timeZoneEnabled && getString(oldCJ.Spec.TimeZone) != getString(newCJ.Spec.TimeZone)) {
```

Comment:
You could replace `getString` comparison with `k8s.io/utils/pointer.StringEqual`, and `getString` with `k8s.io/utils/pointer.StringDeref`